rest/route/github.go

```diff
@@ -370,26 +372,39 @@ func (gh *githubHookApi) handleGitTag(ctx context.Context, event *github.PushEve
 	}
 	catcher := grip.NewBasicCatcher()
 	for _, pRef := range projectRefs {
-		// if a version for this revision exists for this project, add tag
-		existingVersion, err := gh.sc.FindVersionByProjectAndRevision(pRef.Id, hash)
+		var existingVersion *model.Version
+		err = util.Retry(
+			ctx,
+			func() (bool, error) {
+				// If a version for this revision exists for this project, add tag
+				// Retry in case a commit and a tag are pushed at around the same time, and the version isn't ready yet
+				existingVersion, err = gh.sc.FindVersionByProjectAndRevision(pRef.Id, hash)
+				if err != nil {
+					return true, err
+				}
+				if existingVersion == nil {
+					return true, errors.New("no existing version to add tag to")
+				}
+				return false, nil
+			}, 5, time.Second, time.Second*10)
+
 		if err != nil {
 			catcher.Add(errors.Wrapf(err, "problem finding version for project '%s' with revision '%s' to add tag '%s'",
 				pRef.Id, hash, tag.Tag))
-			continue
-		}
-		if existingVersion == nil {
-			grip.Debug(message.Fields{
-				"source":  "github hook",
-				"message": "no version to add tag to",
-				"ref":     event.GetRef(),
-				"event":   gh.eventType,
-				"project": pRef.Id,
-				"branch":  pRef.Branch,
-				"owner":   pRef.Owner,
-				"repo":    pRef.Repo,
-				"hash":    hash,
-				"tag":     tag,
-			})
+			if existingVersion == nil {
```

Comment:
There's no reason to log this specifically anymore, since the error "no existing version to add tag to" will be logged at the end since you added it to the catcher, if that makes sense.